lib/ansible/plugins/lookup/ini.py

```diff
@@ -105,11 +91,23 @@ def run(self, terms, variables=None, **kwargs):
             except (ValueError, AssertionError) as e:
                 raise AnsibleError(e)
 
+            # Retrieve file path
             path = self.find_file_in_search_path(variables, 'files', paramvals['file'])
+
+            # Create StringIO later used to parse ini
+            config = StringIO()
+            # Special case for java properties
             if paramvals['type'] == "properties":
-                var = self.read_properties(path, key, paramvals['default'], paramvals['re'])
-            else:
-                var = self.read_ini(path, key, paramvals['section'], paramvals['default'], paramvals['re'])
+                config.write(u'[java_properties]\n')
+                paramvals['section'] = 'java_properties'
+
+            # Open file using encoding
+            contents, show_data = self._loader._get_file_contents(path, encoding=paramvals['encoding'])
```

Comment:
Cool.  So this seems like the right strategy for accessing files on the controller.